plugins/inputs/intel_baseband/intel_baseband.go

```diff
@@ -0,0 +1,222 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux && amd64
+
+package intel_baseband
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	// plugin name. Exposed with all metrics
+	pluginName = "intel_baseband"
+
+	// VF Metrics
+	vfCodeBlocks = "Code Blocks"
+	vfDataBlock  = "Data (Bytes)"
+
+	// Engine Metrics
+	engineBlock = "Per Engine"
+
+	// Socket extensions
+	socketExtension  = ".sock"
+	logFileExtension = ".log"
+
+	// UnreachableSocketBehavior Values
+	unreachableSocketBehaviorError  = "error"
+	unreachableSocketBehaviorIgnore = "ignore"
+
+	defaultAccessSocketTimeout   = config.Duration(time.Second)
+	defaultWaitForTelemetryDelay = config.Duration(50 * time.Millisecond)
+)
+
+var unreachableSocketBehaviors = []string{unreachableSocketBehaviorError, unreachableSocketBehaviorIgnore}
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Baseband struct {
+	// required params
+	SocketPath  string `toml:"socket_path"`
+	FileLogPath string `toml:"log_file_path"`
+
+	//optional params
+	UnreachableSocketBehavior string          `toml:"unreachable_socket_behavior"`
+	SocketAccessTimeout       config.Duration `toml:"socket_access_timeout"`
+	WaitForTelemetryDelay     config.Duration `toml:"wait_for_telemetry_delay"`
+
+	Log      telegraf.Logger `toml:"-"`
+	logConn  *LogConnector
+	sockConn *SocketConnector
+}
+
+func (b *Baseband) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init performs one time setup of the plugin
+func (b *Baseband) Init() error {
+	var err error
+
+	if b.SocketAccessTimeout < 0 {
+		return fmt.Errorf("socket_access_timeout should be positive number or equal to 0 (to disable timeouts)")
+	}
+	if b.WaitForTelemetryDelay < 0 {
+		return fmt.Errorf("wait_for_telemetry_delay should be positive number or equal to 0 (to disable delay)")
+	}
+
+	// Filling default values
+	// Check UnreachableSocketBehavior
+	if len(b.UnreachableSocketBehavior) == 0 {
+		b.UnreachableSocketBehavior = unreachableSocketBehaviorError
+	} else if err = choice.Check(b.UnreachableSocketBehavior, unreachableSocketBehaviors); err != nil {
+		return fmt.Errorf("unreachable_socket_behavior: %w", err)
+	}
```

Comment:
@srebhan
I believe this is very easy:

1. if string is empty - use default
2. otherwise - check if it was set to allowed value
  - if yes - use it
  - if no - return error

It can be be changed to something different but I don't know if it would be easier. If you have some idea, please share :)

Reply:
```suggestion
	switch b.UnreachableSocketBehavior {
	case "":
		b.UnreachableSocketBehavior = "error"
	case "error", "ignore":
		// Valid options, do nothing
	default:
		return fmt.Errorf("unreachable_socket_behavior %q", b.UnreachableSocketBehavior)
	}
```
for me is easier to read...

Reply:
@srebhan I'm convinced :) Changed.